Automattic/wp-calypso
# Gutenframe: Move TinyMCE plugin to Calypso monorepo

Part of #32044.

#### Changes proposed in this Pull Request

* New `tinymce.js` file on the `@automattic/wpcom-block-editor` package containing the TinyMCE plugin that replaces the media modal used on the classic blocks with the Calypso media modal. It's the same logic that the one included on the `gutenberg-wpcom-classic-block-tinymce-plugin.js` file currently maintained in WP.com, but using ES2015+ syntax.
* New `utils.js` file on the `@automattic/wpcom-block-editor` package containing shared utilities.
* Added support for the `tinymce` dependency to the `@automattic/wordpress-external-dependencies-plugin` package, since it is available as global on WordPress environments (added @sirreal as reviewer to check this change).
* Added `tinymce` as dependency of the `@automattic/wpcom-block-editor` package so we don't need to declare it as global anymore. 
* Changed the build script so we generate an additional build bundle for the TinyMCE plugin.

#### Testing instructions

* Build the package: `npx lerna run build --scope='@automattic/wpcom-block-editor'`.
* Upload the `iframe-bridge-server.js` build file (`/apps/wpcom-block-editor/dist/iframe-bridge-server.js`) to your sandbox and overwrite the content of the `gutenberg-wpcom-iframe-bridge.js` file with the content of the build file.
* Upload the `tinymce.js` build file (`/apps/wpcom-block-editor/dist/tinymce.js`) to your sandbox and overwrite the content of the `gutenberg-wpcom-classic-block-tinymce-plugin.js` file with the content of the build file.
* Load `https://wordpress.com/block-editor` and select your sandbox site.
* Insert a classic block.
* Make sure the Calypso media modal appears for managing the media items added to the classic block.
